update/common.go

```diff
@@ -124,33 +128,15 @@ func getAllMiniBlocksWithDst(metaBlock *block.MetaBlock, destShardID uint32) []b
 	return mbHdrs
 }
 
-// CheckDuplicates checks if there are duplicated miniBlocks in the given map
-func CheckDuplicates(headers map[uint32]data.HeaderHandler) error {
-	miniBlocksHashes := make([][]byte, 0)
-	for _, header := range headers {
-		miniBlocksHashes = append(miniBlocksHashes, header.GetMiniBlockHeadersHashes()...)
-	}
-
-	mapMiniBlocksHashes := make(map[string]struct{})
-	for _, miniBlockHash := range miniBlocksHashes {
-		_, duplicatesFound := mapMiniBlocksHashes[string(miniBlockHash)]
-		if duplicatesFound {
-			return ErrDuplicatedMiniBlocksFound
-		}
-
-		mapMiniBlocksHashes[string(miniBlockHash)] = struct{}{}
-	}
-
-	return nil
-}
-
 // CreateBody will create a block body after hardfork import
 func CreateBody(
```

Comment:
at least 3 parameters from this function and from CleanDuplicates are the same: hasher, marshalizer and shardIDs. In this case, I suggest we have a struct that will hold the common requirements as fields and so the argument count in these functions will drop. This struct can then be embedded in the structs that will use this functionality.

Reply:
done

Reply:
not quite done, it is better, can remain as it is for the moment

Reply:
refactored